### PR TITLE
Add more languages to testdata.

### PR DESCRIPTION
**Pull request by zbraniecki**
Decided to add:

* Greek
* Spanish (+Argentina)
* Thai
* Turkish

Based on my conv. with Shane, Greek adds a new alphabet and Southern Europe, Turkish adds Middle East, Spanish is widely used and Argentina adds South America and Thai is a challenging alphabet.

I'm not strongly opinionated on those, but I'd like to get a total of 10 base languages to use in benchmarks, so I hope this will work. We can refine the list as we identify new challenging locales and replace some of those with them.

### CODE REVIEW

**Comment by zbraniecki** [comment]
Hmm, seems like unfortunately the languages I chose don't have complex plural rules. It may be ok (we shouldn't curate for a single component), but maybe there's a locale that could replace one or two of them with some interesting plurals?

File: resources/testdata/Cargo.toml [hunk 1]
```diff
@@ -47,6 +60,14 @@ locales = [
     # - Often the first non-English locale to receive new data in CLDR
     "fr",
 
+    # Thai:
+    # - Complex word breaking
+    "th",
+
+    # Turkish:
+    # - Interesting case-mappings
```
Comment (alerque) [inline resources/testdata/Cargo.toml:68]: This is putting it nicely. Other ‘I’ words you could have used for "interesting" might be "insidious" or "insufferable". It is also nearly "incomprehensible" on the part of Unicode — mind you the Turkish alphabet and language by themselves make perfect sense, but the "ill-advised" re-use of code points for fundamentally different letters in the encoding "inevitably" leads to a nearly "infinite" string of "inappropriate", "inopportune", and even "illegitimate" usage in the wild. Daha fazla diyebilirdim ama İgilizce'de ı harfı içeren kelime olmadığı için zorlanıyorum.

**Review by Manishearth — APPROVED** [review]

File: resources/testdata/Cargo.toml (hunk 1 above)
Comment (alerque) [reply resources/testdata/Cargo.toml]: And now you dropped Turkish? To my knowledge none of the languages you went for in the latest change have the same kind of casing issues. Turkish is a good test for handling that issue, which only a handful of languages have.

Comment (zbraniecki) [reply resources/testdata/Cargo.toml:77]: readded

**Review by sffc — APPROVED** [review]

**Pull request merged**